// react-flexboxgrid/lib/GridRow.js
"use strict";

var _get = require("babel-runtime/helpers/get")["default"];

var _inherits = require("babel-runtime/helpers/inherits")["default"];

var _createClass = require("babel-runtime/helpers/create-class")["default"];

var _classCallCheck = require("babel-runtime/helpers/class-call-check")["default"];

var _defineProperty = require("babel-runtime/helpers/define-property")["default"];

var _extends = require("babel-runtime/helpers/extends")["default"];

var _interopRequireDefault = require("babel-runtime/helpers/interop-require-default")["default"];

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _classnames2 = require("classnames");

var _classnames3 = _interopRequireDefault(_classnames2);

var _flexboxgridDistFlexboxgridCss = require("flexboxgrid/dist/flexboxgrid.css");

var _flexboxgridDistFlexboxgridCss2 = _interopRequireDefault(_flexboxgridDistFlexboxgridCss);

var GridRow = (function (_React$Component) {
  _inherits(GridRow, _React$Component);

  function GridRow() {
    _classCallCheck(this, GridRow);

    _get(Object.getPrototypeOf(GridRow.prototype), "constructor", this).apply(this, arguments);
  }

  _createClass(GridRow, [{
    key: "render",
    value: function render() {
      var _classnames;

      var _props = this.props;
      var reverse = _props.reverse;
      var xsAlign = _props.xsAlign;
      var smAlign = _props.smAlign;
      var mdAlign = _props.mdAlign;
      var lgAlign = _props.lgAlign;
      var xsVAlign = _props.xsVAlign;
      var smVAlign = _props.smVAlign;
      var mdVAlign = _props.mdVAlign;
      var lgVAlign = _props.lgVAlign;
      var styles = _props.styles;
      var tagName = _props.tagName;
      var className = _props.className;
      var children = _props.children;

      if (styles) {
        if (console && console.warn) {
          console.warn("The `styles` prop is deprecated and will be removed in the future. Use the standard `style` prop instead.");
        }
      }

      var classes = (0, _classnames3["default"])((_classnames = {}, _defineProperty(_classnames, _flexboxgridDistFlexboxgridCss2["default"].row, true), _defineProperty(_classnames, _flexboxgridDistFlexboxgridCss2["default"].reverse, reverse), _defineProperty(_classnames, _flexboxgridDistFlexboxgridCss2["default"][xsAlign + "-xs"], xsAlign), _defineProperty(_classnames, _flexboxgridDistFlexboxgridCss2["default"][smAlign + "-sm"], smAlign), _defineProperty(_classnames, _flexboxgridDistFlexboxgridCss2["default"][mdAlign + "-md"], mdAlign), _defineProperty(_classnames, _flexboxgridDistFlexboxgridCss2["default"][lgAlign + "-lg"], lgAlign), _defineProperty(_classnames, _flexboxgridDistFlexboxgridCss2["default"][xsVAlign + "-xs"], xsVAlign), _defineProperty(_classnames, _flexboxgridDistFlexboxgridCss2["default"][smVAlign + "-sm"], smVAlign), _defineProperty(_classnames, _flexboxgridDistFlexboxgridCss2["default"][mdVAlign + "-md"], mdVAlign), _defineProperty(_classnames, _flexboxgridDistFlexboxgridCss2["default"][lgVAlign + "-lg"], lgVAlign), _classnames), className);

      // The name of the variable has to start with an uppercase otherwise JSX
      // would literally create `<element>`.
      var Element = tagName;

      return _react2["default"].createElement(
        Element,
        _extends({}, this.props, {
          className: classes,
          style: this.props.style || styles }),
        children
      );
    }
  }], [{
    key: "propTypes",
    value: {
      reverse: _react.PropTypes.bool,
      xsAlign: _react.PropTypes.string,
      smAlign: _react.PropTypes.string,
      mdAlign: _react.PropTypes.string,
      lgAlign: _react.PropTypes.string,
      xsVAlign: _react.PropTypes.string,
      smVAlign: _react.PropTypes.string,
      mdVAlign: _react.PropTypes.string,
      lgVAlign: _react.PropTypes.string,
      styles: _react.PropTypes.object, // TODO: remove this
      tagName: _react.PropTypes.string
    },
    enumerable: true
  }, {
    key: "defaultProps",
    value: {
      tagName: "div"
    },
    enumerable: true
  }]);

  return GridRow;
})(_react2["default"].Component);

exports["default"] = GridRow;
module.exports = exports["default"];
