// react-flexboxgrid/lib/GridColumn.js
"use strict";

var _get = require("babel-runtime/helpers/get")["default"];

var _inherits = require("babel-runtime/helpers/inherits")["default"];

var _createClass = require("babel-runtime/helpers/create-class")["default"];

var _classCallCheck = require("babel-runtime/helpers/class-call-check")["default"];

var _defineProperty = require("babel-runtime/helpers/define-property")["default"];

var _extends = require("babel-runtime/helpers/extends")["default"];

var _interopRequireDefault = require("babel-runtime/helpers/interop-require-default")["default"];

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _classnames2 = require("classnames");

var _classnames3 = _interopRequireDefault(_classnames2);

var _flexboxgridDistFlexboxgridCss = require("flexboxgrid/dist/flexboxgrid.css");

var _flexboxgridDistFlexboxgridCss2 = _interopRequireDefault(_flexboxgridDistFlexboxgridCss);

var GridColumn = (function (_React$Component) {
  _inherits(GridColumn, _React$Component);

  function GridColumn() {
    _classCallCheck(this, GridColumn);

    _get(Object.getPrototypeOf(GridColumn.prototype), "constructor", this).apply(this, arguments);
  }

  _createClass(GridColumn, [{
    key: "render",
    value: function render() {
      var _classnames;

      var _props = this.props;
      var xs = _props.xs;
      var sm = _props.sm;
      var md = _props.md;
      var lg = _props.lg;
      var xsOffset = _props.xsOffset;
      var smOffset = _props.smOffset;
      var mdOffset = _props.mdOffset;
      var lgOffset = _props.lgOffset;
      var styles = _props.styles;
      var tagName = _props.tagName;
      var className = _props.className;
      var children = _props.children;

      if (styles) {
        if (console && console.warn) {
          console.warn("The `styles` prop is deprecated and will be removed in the future. Use the standard `style` prop instead.");
        }
      }

      var classes = (0, _classnames3["default"])((_classnames = {}, _defineProperty(_classnames, _flexboxgridDistFlexboxgridCss2["default"]["col-xs"], xs && xs === "auto"), _defineProperty(_classnames, _flexboxgridDistFlexboxgridCss2["default"]["col-sm"], sm && sm === "auto"), _defineProperty(_classnames, _flexboxgridDistFlexboxgridCss2["default"]["col-md"], md && md === "auto"), _defineProperty(_classnames, _flexboxgridDistFlexboxgridCss2["default"]["col-lg"], lg && lg === "auto"), _defineProperty(_classnames, _flexboxgridDistFlexboxgridCss2["default"]["col-xs-" + xs], xs && xs !== "auto"), _defineProperty(_classnames, _flexboxgridDistFlexboxgridCss2["default"]["col-sm-" + sm], sm && sm !== "auto"), _defineProperty(_classnames, _flexboxgridDistFlexboxgridCss2["default"]["col-md-" + md], md && md !== "auto"), _defineProperty(_classnames, _flexboxgridDistFlexboxgridCss2["default"]["col-lg-" + lg], lg && lg !== "auto"), _defineProperty(_classnames, _flexboxgridDistFlexboxgridCss2["default"]["col-xs-offset-" + xsOffset], xsOffset), _defineProperty(_classnames, _flexboxgridDistFlexboxgridCss2["default"]["col-sm-offset-" + smOffset], smOffset), _defineProperty(_classnames, _flexboxgridDistFlexboxgridCss2["default"]["col-md-offset-" + mdOffset], mdOffset), _defineProperty(_classnames, _flexboxgridDistFlexboxgridCss2["default"]["col-lg-offset-" + lgOffset], lgOffset), _classnames), className);

      // The name of the variable has to start with an uppercase otherwise JSX
      // would literally create `<element>`.
      var Element = tagName;

      return _react2["default"].createElement(
        Element,
        _extends({}, this.props, {
          className: classes,
          style: this.props.style || styles }),
        children
      );
    }
  }], [{
    key: "propTypes",
    value: {
      xs: _react.PropTypes.oneOfType([_react.PropTypes.string, _react.PropTypes.number]),
      sm: _react.PropTypes.oneOfType([_react.PropTypes.string, _react.PropTypes.number]),
      md: _react.PropTypes.oneOfType([_react.PropTypes.string, _react.PropTypes.number]),
      lg: _react.PropTypes.oneOfType([_react.PropTypes.string, _react.PropTypes.number]),
      xsOffset: _react.PropTypes.number,
      smOffset: _react.PropTypes.number,
      mdOffset: _react.PropTypes.number,
      lgOffset: _react.PropTypes.number,
      styles: _react.PropTypes.object,
      tagName: _react.PropTypes.string
    },
    enumerable: true
  }, {
    key: "defaultProps",
    value: {
      tagName: "div"
    },
    enumerable: true
  }]);

  return GridColumn;
})(_react2["default"].Component);

exports["default"] = GridColumn;
module.exports = exports["default"];
